compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_materialize_encoding.mlir

```diff
@@ -0,0 +1,95 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-codegen-gpu-materialize-device-encoding))" --split-input-file %s | FileCheck %s
+
+//-----------------------------------------------------------------------------
+// 1. MFMA_F32_16x16x4_F32
+//-----------------------------------------------------------------------------
+
+#encoding = #iree_encoding.encoding<operand_index = 1, op_type = matmul, element_types = [f32, f32, f32], original_type = tensor<255x513xf32>,
+                                    user_indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d2)>, affine_map<(d0, d1, d2) -> (d2, d1)>, affine_map<(d0, d1, d2) -> (d0, d1)>],
+                                    round_dims_to = array<i64: 16, 16, 16>>
+#executable_target_rocm_hsaco_fb = #hal.executable.target<"rocm", "rocm-hsaco-fb", {
```

Comment:
We don't need to embed this whole structure in test if you use `--iree-gpu-test-target=cdna3` or something to have it "autopouplate". You'd need to use [`getGPUTargetAttr` utility function](https://github.com/iree-org/iree/blob/b0512e2/compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp#L960-L999).

Reply:
This is a little tricky in the data-tiling setup because the type converter only takes tensor type and executable target attribute. We want an unified API setup for all the backends. The IREE::GPU::TargetAttr is defined separately which makes it hard. Perhaps I can expose the `getCLGPUTarget` method, so I can have better setup for the test. It is also needed in my other prototype. I'll fix it in a follow-up.